notes/benchmarks/benchmarks_kpis.md

```diff
@@ -0,0 +1,71 @@
+# Document and validate KPIs for Deku benchmarks
+
+Open [issue 574](https://github.com/marigold-dev/deku/issues/574)
+
+## Summary
+
+Define KPIs (Key Performance Indicators) to benchmark various parts of Deku (consensus, hashing...)
+
+## Motivation
+
+We want to measure Deku's ability to handle an important number of transactions. This number depends on several factors (number of validators, hardware, network...). We want to establish a protocol to make this measurements and find potential bottlenecks in the project.
+
+## Production
+
+We define the KPIs into 3 states, and by priority order.
+
+### State 1
+
+Focus on getting the number of TPS, hardware configurations, etc.
+ 
+There are two enviroments we will do the benchmark:
+- Local environment existing on a single machine
+- Cloud: run Deku's chain on different hardware configurations
+
+For transactions, there are two kinds of transactions we are taking into account:
+- Normal transaction (A sends 10 xtz to B)
+- Smart contract call (A calls a smart contract C)
+
+For the latter, we need to identify what kind of smart contract we are interested to get the benchmark for. This will require an input from team leader.
+
+The scenario that we have in mind are:
+- Base case: increase the number of transactions (10, 100, 1000, etc.) with the base number of validators (e.g. minimum 4 for Tendermint consensus).
+- Growing transactions, validators: the same pack of growing transactions (10, 100, 1000, etc.) with the growing of number of valiators (4, 6, 8, 10, etc.)
+
+Different hardware configurations (cloud) set up for same pack of the scenario above.
```

Comment:
Related to previous question on hardware: are you testing specific configurations offered by typical cloud providers, or are you targeting independent hardware variables (increasing RAM size, increasing CPU power…)?

Reply:
The second one: For the variables: increase RAM size, CPU power. (if the cloud providers have)